services/web/client/source/class/osparc/desktop/StudyEditor.js

```diff
@@ -376,6 +388,203 @@ qx.Class.define("osparc.desktop.StudyEditor", {
       }
     },
 
+    __getBrotherNodes: function(excludeNodeIds) {
+      const currentModel = this.__workbenchUI.getCurrentModel();
+      const workbench = this.getStudy().getWorkbench();
+
+      let brotherNodesObj = {};
+      if (currentModel.getNodeId) {
+        brotherNodesObj = currentModel.getInnerNodes(false);
+      } else {
+        brotherNodesObj = workbench.getNodes(false);
+      }
+
+      const brotherNodes = [];
+      for (const brotherNodeId in brotherNodesObj) {
+        const index = excludeNodeIds.indexOf(brotherNodeId);
+        if (index === -1) {
+          const brotherNode = workbench.getNode(brotherNodeId);
+          brotherNodes.push(brotherNode);
+        }
+      }
+      return brotherNodes;
+    },
+
+    __getAveragePosition: function(nodes) {
+      let avgX = 0;
+      let avgY = 0;
+      nodes.forEach(node => {
+        avgX += node.getPosition().x;
+        avgY += node.getPosition().y;
+      });
+      avgX /= nodes.length;
+      avgY /= nodes.length;
+      return {
+        x: avgX,
+        y: avgY
+      };
+    },
+
+    __groupSelection: function() {
```

Comment:
I feel that this should probably be a part of the Workbench class instead. How is this being represented in the workbench property of the study? Sorry still gotta actually run the PR

Reply:
You are right. Moving it to workbench.